javascript.go

```diff
@@ -73,22 +79,77 @@ func TitleToMethodName(s string) string {
 func TransformArguments(args Arguments) []SimpleParameter {
 	simpleArgs := []SimpleParameter{}
 	for name, arg := range args {
-        isArray, arrayType := IsArrayParameter(arg)
+        isArray, cType, jsType := IsArrayParameter(arg)
+        desciption := arg.Messages.Title.I18N["en-US"]
         if isArray {
-            simpleArgs = append(simpleArgs, SimpleParameter{Name: name, Type: "Array(t." + arrayType + ")"})
+            simpleArgs = append(simpleArgs, SimpleParameter{Name: name, CadType: cType, JsType: jsType, FclType: "Array(t." + cType + ")", Description: desciption})
         } else {
-            simpleArgs = append(simpleArgs, SimpleParameter{Name: name, Type: arg.Type})
+            jsType := ConvertCadenceTypeToJS(arg.Type)
+            simpleArgs = append(simpleArgs, SimpleParameter{Name: name, CadType: arg.Type, JsType: jsType, FclType: arg.Type, Description: desciption})
         }
 	}
 	return simpleArgs
 }
 
 
-func IsArrayParameter(arg Argument) (bool, string) {
+func IsArrayParameter(arg Argument) (bool, string, string) {
     isArray := arg.Type[0] == '[' && arg.Type[len(arg.Type)-1] == ']'
     if (!isArray) {
-        return isArray, ""
+        return isArray, "", ""
     }
-    return isArray, arg.Type[1 : len(arg.Type)-1]
+    cadenceType := arg.Type[1 : len(arg.Type)-1]
+    jsType := "Array<" + ConvertCadenceTypeToJS(cadenceType) + ">"
+    return isArray, cadenceType, jsType
 }
 
+// ConvertCadenceTypeToJS takes a Cadence type as a string and returns its JavaScript equivalent
+func ConvertCadenceTypeToJS(cadenceType string) string {
+    switch cadenceType {
+    case "Int":
+        return "Number"
+    case "Int8":
+        return "Number"
+    case "Int16":
+        return "Number"
+    case "Int32":
+        return "Number"
+    case "Int64":
+        return "BigInt"
+    case "Int128":
+        return "BigInt"
+    case "Int256":
+        return "BigInt"
+    case "UInt":
+        return "string"
+    case "UInt8":
+        return "string"
+    case "UInt16":
+        return "string"
+    case "UInt32":
+        return "string"
+    case "UInt64":
+        return "string"
+    case "UInt128":
+        return "string"
+    case "UInt256":
+        return "string"
+    case "UFix64":
+        return "string"
+    case "Fix64":
+        return "BigInt"
```

Comment:
Think this should be string along with the other number types. FCL takes strings for numbers. But that is a potential problem if someone else has a library with a different way of handling them. May need to keep a mapping for each library.

Reply:
Good point about supporting multiple js clients. Not sure there are any other js libs that don't use fcl. If there are some I can fold them in.

Reply:
reorganizing code to pass in the generator method so flow-cli will pass in the generator method, this will open up possibility to use anyone's generate lib.